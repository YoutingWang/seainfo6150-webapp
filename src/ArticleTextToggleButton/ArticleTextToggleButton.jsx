import React from "react";
import styles from "./ArticleTextToggleButton.module.css";

const ArticleTextToggleButton = ({buttonText, onClick}) => {
  return (
    <button className = {styles.toggleButton} onClick = {onClick}>
        {buttonText}
    </button>
  );
};

export default ArticleTextToggleButton;