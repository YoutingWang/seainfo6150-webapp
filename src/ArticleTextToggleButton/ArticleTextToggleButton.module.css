.toggleButton{
    background-color:#c92e2e;
    color: white;
    border: 3px solid gray;
    box-shadow: 3px 3px black;
    border-radius: 10px;
    padding: 4px;
    font-size: 16px;
    width: 100%;
}