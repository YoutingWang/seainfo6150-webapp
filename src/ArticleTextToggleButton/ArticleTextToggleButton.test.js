import React from "react";
import ArticleTextToggleButton from "./ArticleTextToggleButton";
import {
    fireEvent
} from "@testing-library/react";

describe("ArticleTextToggleButton tests", () => {
    it("renders correctly when button is Show more", () => {
        const {
            container
        } = render( < ArticleTextToggleButton buttonText = "Show more" / > );
        expect(container).toMatchSnapshot();
    });

    it("renders correctly when button is Show less", () => {
        const {
            container
        } = render( < ArticleTextToggleButton buttonText = "Show less" / > );
        expect(container).toMatchSnapshot();
    });

    it("renders correctly after Show more button is clicked", () => {
        const {
            container,
            getByText
        } = render( <
            ArticleTextToggleButton buttonText = "Show more" / >
        );
        const button = getByText("Show more");
        fireEvent.click(button);
        expect(container).toMatchSnapshot();
    });

    it("renders correctly after Show less button is clicked", () => {
        const {
            container,
            getByText
        } = render( <
            ArticleTextToggleButton buttonText = "Show less" / >
        );
        const button = getByText("Show less");
        fireEvent.click(button);
        expect(container).toMatchSnapshot();
    });
});