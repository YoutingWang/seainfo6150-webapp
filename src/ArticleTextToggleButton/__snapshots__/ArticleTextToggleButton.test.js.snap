// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleTextToggleButton tests renders correctly after Show less button is clicked 1`] = `
<div>
  <button
    class="toggleButton"
  >
    Show less
  </button>
</div>
`;

exports[`ArticleTextToggleButton tests renders correctly after Show more button is clicked 1`] = `
<div>
  <button
    class="toggleButton"
  >
    Show more
  </button>
</div>
`;

exports[`ArticleTextToggleButton tests renders correctly when button is Show less 1`] = `
<div>
  <button
    class="toggleButton"
  >
    Show less
  </button>
</div>
`;

exports[`ArticleTextToggleButton tests renders correctly when button is Show more 1`] = `
<div>
  <button
    class="toggleButton"
  >
    Show more
  </button>
</div>
`;
